test/integration/hcpvaultsecretsapp_integration_test.go

```diff
@@ -0,0 +1,405 @@
+				// test that the custom resources can be deleted before tf destroy
+				// removes the k8s namespace
+				assert.Nil(t, crdClient.Delete(ctx, c))
+			}
+
+			exportKindLogs(t)
+
+			// Clean up resources with "terraform destroy" at the end of the test.
+			terraform.Destroy(t, tfOptions)
+			os.RemoveAll(tempDir)
+
+			// Undeploy Kustomize
+			if !testWithHelm {
+				k8s.KubectlDeleteFromKustomize(t, k8sOpts, kustomizeConfigPath)
+			}
+		} else {
+			t.Logf("Skipping cleanup, tfdir=%s", tfDir)
+		}
+	})
+
+	// Run "terraform init" and "terraform apply". Fail the test if there are any errors.
+	terraform.InitAndApply(t, tfOptions)
+
+	if skipCleanup {
+		// save vars to re-run terraform, useful when SKIP_CLEANUP is set.
+		b, err := json.Marshal(tfOptions.Vars)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if err := os.WriteFile(
+			filepath.Join(tfOptions.TerraformDir, "terraform.tfvars.json"), b, 0o644); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	b, err := json.Marshal(terraform.OutputAll(t, tfOptions))
+	require.Nil(t, err)
+
+	var outputs hcpVSOutputs
+	require.Nil(t, json.Unmarshal(b, &outputs))
+
+	auths := []*secretsv1beta1.HCPAuth{
+		{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      consts.NameDefault,
+				Namespace: operatorNS,
+			},
+			Spec: secretsv1beta1.HCPAuthSpec{
+				Method:         hcp.ProviderMethodServicePrincipal,
+				OrganizationID: hcpOrganizationID,
+				ProjectID:      hcpProjectID,
+				ServicePrincipal: &secretsv1beta1.HCPAuthServicePrincipal{
+					SecretRef: outputs.SPSecretName,
+				},
+			},
+		},
+	}
+
+	create := func(o ctrlclient.Object) {
+		require.Nil(t, crdClient.Create(ctx, o))
+		created = append(created, o)
+	}
+
+	for _, o := range auths {
+		create(o)
+	}
+
+	tests := []struct {
+		name         string
+		expectedData map[string][]byte
+		updateData   map[string][]byte
+	}{
+		{
+			name: "basic",
+			expectedData: map[string][]byte{
+				"foo": []byte(`baz`),
+			},
+		},
+		{
+			name: "updates",
+			expectedData: map[string][]byte{
+				"foo": []byte(`baz`),
+			},
+			updateData: map[string][]byte{
+				"foo": []byte(`buz`),
+			},
+		},
+	}
+
+	createHVSSecrets := func(t *testing.T, appName string, data map[string][]byte) error {
+		t.Helper()
+		var errs error
+		for k, v := range data {
+			_, err := hvsClient.CreateAppKVSecret(
+				hvsclient.NewCreateAppKVSecretParams().WithContext(ctx).
+					WithAppName(appName).
+					WithLocationOrganizationID(hcpOrganizationID).
+					WithLocationProjectID(hcpProjectID).
+					WithBody(
+						hvsclient.CreateAppKVSecretBody{
+							Name:  k,
+							Value: string(v),
+						}),
+				nil)
+			if err != nil {
+				errs = errors.Join(err)
+			}
+		}
+		return errs
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt := tt
+			t.Parallel()
+
+			var objsCreated []ctrlclient.Object
+
+			sum := sha256.Sum256([]byte(fmt.Sprintf("%s-%s", outputs.NamePrefix, t.Name())))
+			appName := fmt.Sprintf("vso-test-%x", sum[:8])
```

Comment:
I wonder if it would still be good to have the test name in the appName, might make it easier to debug on the HCP Vault Secrets side.

Reply:
Good point, I did have it like that but HVS has name restriction of 32 chars.

Reply:
We were including that bit in the app's description.  I extended that a bit in 4298d90e8471bbac5c59f4ff45f30432c4eb762b